DESIGN_DOC.md

```diff
@@ -0,0 +1,99 @@
+We would need to set up an appropriate hook on our side to recieve these notifications. Our hook will recieve notifications for both newly created events
+and updated events - this will affect our preprocessing of these events which will be described later.
+
+### Disaster monitoring
+Disaster events can be queried from some provided APIs [4] [5]. Which API we use is not particularly relevant at the moment - the only difference will be in the way we determine which disaster events affect which calendar events (more on that later). We can set up a cron job that will query the disaster API on a regular basis and store received events into the database.
+
+### Matching disasters with calendar events
+A reasonable way to determine whether some disaster event affects any of the calendar events would be to select all disaster events at a specified radius from the calendar event. This method is appropriate for the [4] disaster API, which provides precise coordinates of the former.
+
+A different approach would be to match a general location of both the disaster event and the calendar event. For example, we can consider some calendar event to be affected by the disaster if both events occur in the same country. This method is suitable for the [5] API.
+
+If the first method is selected, we will need to determine the precise coordinates of the calendar event since the calendar API only provides a textual representation of the address. We can use Google's Maps API to achieve that. We can also use this API to determine the distance between two points on the globe.
+
+If we choose the second approach, we would still need to use the Maps API to determine the country of the address. However, this time it would be more complicated to determine if two events are from the same location since we can have two different strings describing the same country (e.g. `USA` and `US` or `US` and `United States of America`). Thus, I think that the first approach is the better option.
+
+### User-specified configurations
+It might be a good idea to allow users to configure various parameters of our API. These might include:
+1. The time interval for the Calendar events that can receive warnings from our API.
+Users can create calendar events for any point in time. It doesn't make any sense to produce warnings for events from the past since those have already happened. It also doesn't make much sense to produce warnings for the events in the far future since a lot can change before those events occur. Thus, the user can specify the amount of days to determine which calendar events should be monitored. We call this `Calendar time-frame`. More precisely, given the calendar time-frame `T` and some calendar event `E`, event `E` will only be monitored if it's time of arrival is less that or equal to `now + T`. Note that, since the user can create calendar events for arbitrary points in time, some of the created events might become "monitorable" in the future.
```

Comment:
I don't think it is needed. We can have our warning cron job to run every our, filter out all events and planned calendar events that match (e.g. not outdated, still planned), and then generate the notifications. All calendar events will be stored in one table for simplicity and we can easily determine when we want to send notifications (e.g. can start send notifs 1 week before the start of event, and once the event is going).